src/index.ts

```diff
@@ -153,9 +153,23 @@ function preactPlugin({
 				build: {
 					rollupOptions: {
 						onwarn(warning, warn) {
-							// Silence Rollup's module-level directive warnings -- they're likely
-							// to all come from `node_modules` (RSCs) and won't be actionable.
-							if (warning.code === "MODULE_LEVEL_DIRECTIVE") return;
+							// Silence Rollup's module-level directive warnings re:"use client".
+							// They're likely to come from `node_modules` and won't be actionable.
+							if (
+								warning.code === "MODULE_LEVEL_DIRECTIVE" &&
+								warning.message.includes("use client")
```

Comment:
I figure copying Vite's React plugin and restricting this to only `use client` is a good idea -- users should be warned if a `use server` shows up somewhere.